### ISSUE TITLE
SOLVED - Problems setting up bootstrap for Ubuntu channel

### ISSUE THREAD

**Issue by TimShaw61** [uyuni-project/uyuni#1449]
uyuni 4.0.2
I'm following the client configuration guide and have setup channels for ubuntu 18.04 as per the docs.  All these channels are synced and contain packages.  I want to setup the bootstrap so I run:

`mgr-create-bootstrap-repo --with-custom-channels`

as per the docs and get a list of channels to select:

```
1. OES2018-SP1-x86_64
2. SLE-12-SP3-x86_64
3. centos-7-x86_64
4. ubuntu-16.04-amd64
5. ubuntu-18.04-amd64
Enter a number of a product label: 
```

Selecting 5 gives:

```
WARNING: no parent channel found. Execute without using custom channels
Creating bootstrap repo for ubuntu-18.04-amd64

Error: Too few arguments for command 'includedeb'!
Syntax: reprepro [--delete] includedeb <distribution> <.deb-file>
There have been errors!
Error creating bootstrap repo.
```

debug mode reveals that the script could not find any packages are found from the list of those it is looking for yet the packages are present in the channel repositories 

It looks like the problem is that the parent channel can't be found. Querying the database shows the parent channel label but when this is specified  on the command line, the script reports it isn't found:

```
psql -U uyuni -d uyuni -c 'select label
> from rhnChannel
> where parent_channel IS NULL
> and org_id IS NOT NULL'
Password for user uyuni: 
             label             
-------------------------------
 centos7-x86_64
 ubuntu-18.04-pool-amd64-uyuni
(2 rows)
```

then...

```
mgr-create-bootstrap-repo -d -c ubuntu-18.04-amd64 --with-custom-channels --with-parent channel=ubuntu-18.04-pool-amd64-uyuni
'ubuntu-18.04-pool-amd64-uyuni' not found in existing parent channel options '[]'
```

I've tried every possible permutation of command line option I can think of and still no joy.  Any ideas as to what I'm doing wrong?

**Comment by mbologna** [comment]
Is this an upgraded server? Have you run `spacewalk-schema-upgrade`?

**Comment by TimShaw61** [comment]
No - its a fresh installation.  I was able to create other bootstrap repos - for centos7 for example.  There seems to be a problem with mgr-create-bootstrap-repo script not identifying the parent channel from the database

**Comment by TimShaw61** [comment]
OK - problem solved.  The mgr_bootstrap_data.py file in the released 4.0.2 code did not contain the repository information for the base channel 'ubuntu-18.04-pool-amd64-uyuni'.  Updating this file from github allowed the bootstrap repo to be created properly.

**Comment by TimShaw61** [comment]
Solved as described above
